### DISCUSSION TITLE
nng 1.6.0 - prerelease dependency

### DISCUSSION THREAD

**Discussion by nviets** [r-lib/nanonext#6] [Q&A] [answered]
Thanks for updating nanonext's build because it made it easier to build with a system nng. Why was it necessary to use a prerelease, instead of 1.5.2?

**Comment by shikokuchuo** [comment, marked as answer]
nanonext takes advantage of the following commit in upstream NNG (nanomsg/nng@722bf46), which was designed to make it much easier to query completion status of an aio (and helps make our treatment of aios both fast and intuitive). The pre-release status does not necessarily mean it is less stable, the author just hasn't decided to make the next release yet.

**Reply by nviets** [reply]
Thanks! Do you know if nng plans to make a release soon? I see a lot of activity on the project, but there hasn't been a release in over a year.

**Reply by shikokuchuo** [reply]
I have asked the question for you. Will let you know once I hear back.

**Reply by shikokuchuo** [reply]
To keep you in the loop, the reply is positive and he's already worked through a couple of PRs. Possibly next week, but should hopefully be soon in any case.

**Reply by nviets** [reply]
Thanks - no rush. I only noticed the prerelease dependency because I recently added [nng](https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/libraries/nng/default.nix) to nixpkgs to support [nanonext](https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/r-modules/default.nix#L364) and mirai in the R packaging system there. Love you work - thank you!

**Reply by shikokuchuo** [reply]
FYI NNG's maintainer is preparing for the next release. As of [d2d72d9](https://github.com/shikokuchuo/nanonext/commit/d2d72d9c9013d32902ce01f211a6c3f43dda132d), the bundled source is updated to what is considered the alpha, for final testing.

**Reply by nviets** [reply]
Thanks for the heads up! I'll check it out and update nixpkgs when it's released.

**Reply by shikokuchuo** [reply]
Hi @nviets unfortunately there is still work to be done in the alpha NNG library version, hence the next release may take some more time. However, I have taken the step to make `nanonext` back compatible with existing NNG releases >= 1.5 (see the discussion at #10). The current dev version is already compatible, as will the next CRAN version.

**Reply by shikokuchuo** [reply]
Hi @nviets just giving you a headsup - NNG 1.6.0 has finally been released!

**Reply by nviets** [reply]
Exciting news! Already have a PR open in nixpkgs: https://github.com/NixOS/nixpkgs/pull/270351. Will nanonext from CRAN work with this release?

**Reply by shikokuchuo** [reply]
Fantastic! It will definitely work - in fact some of G D'Amore's final commits pre-release save nanonext some workarounds making the code more efficient. I've already updated nanonext to bundle the updated source for those without system libs.

**Reply by nviets** [reply]
Thanks for keeping me posted. Have a look at the nixpkgs project sometime. It pairs naturally with the targets/crew/nanonext infrastructure you have led to scale reproducible on HPC infrastructure (which happens to be my use case). We've been clustermq addicts for years, which is why we're so excited to see what you've done with nng!
